--- server.cc
+++ /src/boringssl/tool/server.cc
@@ -77,6 +77,14 @@
         "Print debug information about the handshake",
     },
     {
+	"-verify-ca-file", kOptionalArgument,
+	"File containing CAs trusted for client certificate authentication",
+    },
+    {
+	"-require-client-cert", kBooleanArgument,
+	"Require client certificate and verify it",
+    },
+    {
         "-require-any-client-cert", kBooleanArgument,
         "The server will require a client certificate.",
     },
@@ -309,6 +317,17 @@
 
   if (args_map.count("-debug") != 0) {
     SSL_CTX_set_info_callback(ctx.get(), InfoCallback);
+  }
+
+  if (args_map.count("-verify-ca-file") != 0) {
+    if(!SSL_CTX_load_verify_locations(ctx.get(), args_map["-verify-ca-file"].c_str(), nullptr)){
+      fprintf(stderr, "error loading ca file\n");
+    }
+  }
+
+  if (args_map.count("-require-client-cert") != 0) {
+    SSL_CTX_set_verify(
+  	ctx.get(), SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, nullptr);
   }
 
   if (args_map.count("-require-any-client-cert") != 0) {
